x-pack/plugins/security_solution/server/lib/detection_engine/signals/utils.ts

```diff
@@ -606,7 +606,15 @@ export const createSearchAfterReturnTypeFromResponse = ({
   timestampOverride: TimestampOverrideOrUndefined;
 }): SearchAfterAndBulkCreateReturnType => {
   return createSearchAfterReturnType({
-    success: searchResult._shards.failed === 0,
+    success:
+      searchResult._shards.failed === 0 ||
+      searchResult?._shards?.failures?.every((failure) => {
+        return (
+          failure.reason?.reason === 'No mapping found for [@timestamp] in order to sort on' ||
+          failure.reason?.reason ===
+            `No mapping found for [${timestampOverride}] in order to sort on`
+        );
+      }),
```

Comment:
```ts
searchResult?._shards?.failures?.every((failure) => {
        return (
          failure.reason?.reason === 'No mapping found for [@timestamp] in order to sort on' ||
          failure.reason?.reason ===
            `No mapping found for [${timestampOverride}] in order to sort on`
        );
      }),
```

Is this logic correct? If you have for all failures these error messages then it returns `success: true`?

Maybe that's what you want, can we have a unit test for this condition within the `utils.test.ts`?

Reply:
yeah I will absolutely add a unit test for this. This logic should prevent the rule from entering an error state if every error it receives is either "no @timestamp field" or "not timestampOverride field". The reason we still want to the rule status to be succeeded with these errors is because we have already written a "partial failure" status before the rule executes the search after / bulk create. So errors that tell us what we already wrote a partial failure status for should not fail the rule. If there are errors that are not related to missing timestamp fields then we want those errors to be the priority when displayed on the rule details page and show the rule as "failing" and not in a "partial failure" state. I'll add a few unit tests around this. Good catch thank you!